assignments/prototypes.js

```diff
@@ -32,79 +52,151 @@
-  console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+const mage = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 1,
+    height: 1
+  },
+  healthPoints: 5,
+  name: "Bruce",
+  team: "Mage Guild",
+  weapons: ["Staff of Shamalama"],
+  language: "Common Tongue"
+});
+
+const swordsman = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 2,
+    height: 2
+  },
+  healthPoints: 15,
+  name: "Sir Mustachio",
+  team: "The Round Table",
+  weapons: ["Giant Sword", "Shield"],
+  language: "Common Tongue"
+});
+
+const archer = new Humanoid({
+  createdAt: new Date(),
+  dimensions: {
+    length: 1,
+    width: 2,
+    height: 4
+  },
+  healthPoints: 10,
+  name: "Lilith",
+  team: "Forest Kingdom",
+  weapons: ["Bow", "Dagger"],
+  language: "Elvish"
+});
+
+console.log(mage.createdAt); // Today's date
+console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+console.log(swordsman.healthPoints); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.team); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(mage.takeDamage()); // Bruce took damage.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+
+// Stretch task:
+// * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.
+// * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+// * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+Humanoid.prototype.attack = function(opponent) {
+  opponent.healthPoints--;
+  if (opponent.healthPoints <= 0) {
+    return opponent.destroy();
+  } else {
+    return `${opponent.takeDamage()} Health Points: ${opponent.healthPoints}`;
+  }
+};
+
+function Villain(value1) {
+  Humanoid.call(this, value1);
+}
+Villain.prototype = Object.create(Humanoid.prototype);
+
+function Hero(value2) {
+  Humanoid.call(this, value2);
+}
+Hero.prototype = Object.create(Humanoid.prototype);
+
+const viggo = new Villain({
+  createdAt: new Date(),
+  dimensions: {
+    length: 4,
+    width: 4,
+    height: 4
+  },
+  healthPoints: 10,
+  name: "Viggo",
+  team: "Dragon Hunters",
+  weapons: ["Sword", "Dagger"],
+  language: "Unknown"
+});
+
+const hiccup = new Hero({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 3,
+    height: 3
+  },
+  healthPoints: 10,
+  name: "Hiccup",
+  team: "Dragon Riders",
+  weapons: ["Fire Sword", "Night Fury Dragon"],
+  language: "Old Norse"
+});
+
+function fight(hero, villain) {
+  // both players have life, attack
+  while (hero.healthPoints > 0 && villain.healthPoints > 0) {
```

Comment:
Great job using a while loop and incorporating Math.random() to decide which character attacks.